lib/viewmodels/comment_view_model.dart

```diff
@@ -0,0 +1,61 @@
+import "dart:async";
+
+import "package:collection/collection.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/ui/comment_post.dart";
+import "package:proxima/services/database/comment_repository_service.dart";
+import "package:proxima/services/database/user_repository_service.dart";
+
+/// This view model is used to fetch the comments of a post.
+/// It fetches the comments under the post with the id [arg] and returns
+/// a list of [CommentPost] objects to be displayed.
+class CommentViewModel
+    extends AutoDisposeFamilyAsyncNotifier<List<CommentPost>, PostIdFirestore> {
+  @override
+  Future<List<CommentPost>> build(PostIdFirestore arg) async {
```

Comment:
I completely agree with you on that.
However, doing so yields a warning from the dart formatting because this is the parameter of an overriden method. The warning is the following:

```dart
The parameter name 'postId' doesn't match the name 'arg' in the overridden method.
Try changing the name to 'arg'.dart[avoid_renaming_method_parameters](https://dart.dev/lints/avoid_renaming_method_parameters)
```

Thus I need to keep the parameter named `arg`.

Reply:
There is maybe a way to suppress this warning locally, would it make sense? What do you think?

Reply:
We can indeed remove the warning by adding `// ignore: avoid_renaming_method_parameters`.
However, I am not sure that this is really a good practice to do so.

I personnaly prefer to avoid ignoring the warning and to keep the parameter named `arg`.
Also, I realised that the class exposes `arg` as an attribute that we can access anywhere in the class methods. And we cannot rename this one.

Reply:
Yes, I agree ignoring warnings is a dangerous habit. Let's keep this as you did! :)
Do you think a small comment justifying this variable name would make sense?

Reply:
Yes, that's a good idea !
Done in d9114e92bda04e2000dbdbb9633fbffb071504d5.